src/UI/TextArea.cpp

```diff
@@ -260,6 +281,39 @@ void TextArea::_calculate()
         {
             symbol.setX(symbol.x() + xOffset);
             symbol.setY(symbol.y() + yOffset);
+            //outline symbols
+            if(_outline)
+            {
+                std::shared_ptr<Font> outlineFont = ResourceManager::font(symbol.font()->filename(), _outlineColor);
+                
+                auto outlineSymbolLeft = new TextSymbol(symbol.chr(), symbol.x()-1, symbol.y());
+                auto outlineSymbolTopLeft = new TextSymbol(symbol.chr(), symbol.x()-1, symbol.y()-1);
+                auto outlineSymbolBottomLeft = new TextSymbol(symbol.chr(), symbol.x()-1, symbol.y()+1);
+                auto outlineSymbolRight = new TextSymbol(symbol.chr(), symbol.x()+1, symbol.y());
+                auto outlineSymbolTopRight = new TextSymbol(symbol.chr(), symbol.x()+1, symbol.y()-1);
+                auto outlineSymbolBottomRight = new TextSymbol(symbol.chr(), symbol.x()+1, symbol.y()+1);
+                auto outlineSymbolBottom = new TextSymbol(symbol.chr(), symbol.x(), symbol.y()-1);
+                auto outlineSymbolTop = new TextSymbol(symbol.chr(), symbol.x(), symbol.y()+1);
+                
+                outlineSymbolLeft->setFont(outlineFont);
+                outlineSymbolTopLeft->setFont(outlineFont);
+                outlineSymbolBottomLeft->setFont(outlineFont);
+                outlineSymbolRight->setFont(outlineFont);
+                outlineSymbolTopRight->setFont(outlineFont);
+                outlineSymbolBottomRight->setFont(outlineFont);
+                outlineSymbolBottom->setFont(outlineFont);
+                outlineSymbolTop->setFont(outlineFont);
+                
+                _symbols.push_back(*outlineSymbolLeft);
+                _symbols.push_back(*outlineSymbolTopLeft);
+                _symbols.push_back(*outlineSymbolBottomLeft);
+                _symbols.push_back(*outlineSymbolRight);
+                _symbols.push_back(*outlineSymbolTopRight);
+                _symbols.push_back(*outlineSymbolBottomRight);
+                _symbols.push_back(*outlineSymbolBottom);
+                _symbols.push_back(*outlineSymbolTop);
```

Comment:
Simply putting repeated stuff in one function would do the job I think, on "client-side" you just tell which offsets you need for each outline. That's DRY principle :)